### PR TITLE
AVM: Derive looser, but more principled, checks of txn max size

### PR DESCRIPTION
**Pull request by jannotti**
Better checks on max txn size that will work better as we make changes.

### CODE REVIEW

File: node/node_test.go [hunk 1]
```diff
@@ -805,11 +807,48 @@ func TestMaxSizesCorrect(t *testing.T) {
...
+
+	// we actually have to hold 16 txns, but in the case of multiple txns in a
+	// group, none can be stateproofs. So derive maxMinusSP, which is a per txn
+	// size estimate that excludes stateproof fields.
+	spTxnSize := uint64(csp.StateProofMaxSize() + stateproofmsg.MessageMaxSize())
+	maxMinusSP := maxCombinedTxnSize - spTxnSize
+	require.Greater(t, txTagMax, 16*maxMinusSP)
+	// when we do logisig pooling, 16*maxMinusSP may be a large overshoot, since
+	// it will assume we can have a big logicsig in _each_ of the 16.  It
+	// probbaly won't matter, since stateproof will still swamp it.  But if so,
```
Comment (jasonpaulos) [inline node/node_test.go:846]:
```suggestion
	// probably won't matter, since stateproof will still swamp it.  But if so,
```

**Review by jasonpaulos — DISMISSED** [review]
Makes sense to me, though I have some minor comments

File: node/node_test.go [hunk 2]
```diff
@@ -805,11 +807,48 @@ func TestMaxSizesCorrect(t *testing.T) {
...
+	// stateproof txn.
+	txTagMax := protocol.TxnTag.MaxMessageSize()
+
+	// SignedTxnMaxSize() is an overestimate of a single transaction because it
+	// includes fields from all the different types of signatures, and types of
+	// transactions. First, we remove the aspects of the overestimate that come
+	// from the multiple signature types.
+	maxCombinedTxnSize := uint64(transactions.SignedTxnMaxSize())
+	// subtract out the two smaller signature sizes (logicsig is biggest, it can *contain* the others)
+	maxCombinedTxnSize -= uint64(crypto.SignatureMaxSize() + crypto.MultisigSigMaxSize())
```
[the remark this answers is not in the document]
Comment (jannotti) [reply node/node_test.go:826]: That makes sense, I didn't pay attention and thought I was looking at generated code.

[the remark this answers is not in the document]
Comment (jannotti) [reply node/node_test.go:826]: I take it back. It is generated code.

**Review by jasonpaulos — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]
Looks reasonable to me

**Review by gmalouf — APPROVED** [review]

**Pull request merged**